### PR TITLE
VIDCS-869: Fix crash when toggling video

### PR DESCRIPTION
**Pull request by vona-ben**
#### What is this PR doing?
Fixes an issue where crashing may occur when toggling video

#### How should this be manually tested?
To reproduce, follow the steps detailed in https://github.com/opentok/opentok-android-sdk-samples/issues/461 

To confirm fix, checkout this branch and then follow the same steps as before 🎉

### CODE REVIEW

**Comment by juliobecerragomez** [comment]
Has this been updated based on our camera2 capturer?

File: Basic-Video-Capturer-Camera-2-Java/app/src/main/java/com/tokbox/sample/basicvideocapturercamera2/MirrorVideoCapturer.java [hunk 1]
```diff
@@ -369,93 +381,121 @@ private int startCameraCapture() {
...
+            /*
+            executeAfterCameraOpened = () -> {
+                cameraState = CameraState.CLOSING;
+                if (camera != null) {
+                    camera.close();
+                }
+            };
 
+             */
+            executeAfterCameraOpened = null;
```
Comment (juliobecerragomez) [inline Basic-Video-Capturer-Camera-2-Java/app/src/main/java/com/tokbox/sample/basicvideocapturercamera2/MirrorVideoCapturer.java:466]: If  I remember well, this code should not be commented out, and the null assignment removed.

File: Basic-Video-Capturer-Camera-2-Java/app/src/main/java/com/tokbox/sample/basicvideocapturercamera2/MirrorVideoCapturer.java [hunk 2]
```diff
@@ -102,126 +98,133 @@ private enum CameraState {
...
-                    Log.d(TAG,"CaptureSession onConfigured");
-
+                    Log.d(TAG,"CameraCaptureSession.StateCallback onConfigured() enter.");
                     try {
                         cameraState = CameraState.CAPTURE;
                         captureSession = session;
-                        captureRequest = captureRequestBuilder.build();
-                        captureSession.setRepeatingRequest(captureRequest, captureNotification, cameraThreadHandler);
+                        CaptureRequest captureRequest = captureRequestBuilder.build();
+                        captureSession.setRepeatingRequest(captureRequest, captureNotification, null);
```
Comment (goncalocostamendes) [inline Basic-Video-Capturer-Camera-2-Java/app/src/main/java/com/tokbox/sample/basicvideocapturercamera2/MirrorVideoCapturer.java:202]: Why are we no longer passing cameraThreadHandler?

File: Basic-Video-Capturer-Camera-2-Java/app/src/main/java/com/tokbox/sample/basicvideocapturercamera2/MirrorVideoCapturer.java [hunk 3]
```diff
@@ -233,17 +236,18 @@ public void onCaptureStarted(CameraCaptureSession session, CaptureRequest reques
 
     /* caching of camera characteristics & display orientation for performance */
     private static class CameraInfoCache {
-        private CameraCharacteristics info;
-        private boolean frontFacing = false;
-        private int sensorOrientation = 0;
+        private final CameraCharacteristics   info;
+        private boolean                 frontFacing = false;
```
Comment (goncalocostamendes) [inline Basic-Video-Capturer-Camera-2-Java/app/src/main/java/com/tokbox/sample/basicvideocapturercamera2/MirrorVideoCapturer.java:240]: nit: identation

**Review by goncalocostamendes — COMMENTED** [review]
Left one question

**Comment by vona-ben** [comment]
@juliobecerragomez @goncalocostamendes Those changes have been taken from the updated native SDK. I haven't made any modifications other than renaming variables and fixing the logs.

**Comment by goncalocostamendes** [comment]
> @juliobecerragomez @goncalocostamendes Those changes have been taken from the native SDK code. I haven't made any modifications other than renaming variables and fixing the logs. I think @jintgeorge would be best to reply to the comment here (cc @v-kpheng )

Yes, just confirmed it is the same as native SDK. Approving

**Review by goncalocostamendes — APPROVED** [review]
LGTM!

**Comment by juliobecerragomez** [comment]
@vona-ben So at least we should fix the commented out code. I do not think it is ok to show it as part of the linux sample apps.

File: Basic-Video-Capturer-Camera-2-Java/app/src/main/java/com/tokbox/sample/basicvideocapturercamera2/MirrorVideoCapturer.java (hunk 1 above)
Comment (jintgeorge) [reply Basic-Video-Capturer-Camera-2-Java/app/src/main/java/com/tokbox/sample/basicvideocapturercamera2/MirrorVideoCapturer.java:466]: I think the same. Not sure why this got commented out. May be we can check the history when this change was made.

**Comment by vona-ben** [comment]
> @vona-ben So at least we should fix the commented out code. I do not think it is ok to show it as part of the linux sample apps.

Done

**Review by goncalocostamendes — APPROVED** [review]
+1

**Review by jintgeorge — APPROVED** [review]
LGTM

**Review by juliobecerragomez — COMMENTED** [review]
LGTM!

**Pull request merged**